code/xcvm/SPEC.md

```diff
@@ -88,13 +90,13 @@ BridgeSecurity ::=
 
 Outgoing messages are routed based on bridge security, or by specifying the bridge contract directly.
 
-Each XCVM execution has access to its message origin and can be configured to deny execution depending on the address or security level:
+Each XCVM execution has access to its message `MessageOrigin` and can be configured to deny execution depending on the address or security level:
 
-```
+```ebpf
 MessageOrigin ::=
     IBC
     | XCM
-    | OTP bytes BridgeSecurity
+    | (OTP, bytes, BridgeSecurity)
```

Comment:
Actually in pseudo-Backus–Naur form, this would be without commas and parens

```suggestion
    | OTP bytes BridgeSecurity
```

Reply:
there was case in the end of document with | (A, B).

so will remove than there?

Reply:
```
Header ::= Hash Number
Data ::= 
    Assets
    | (Result, IP)
    | bytes
```

This one,

So it had comma, and grouping. 

Afaik grouping is useful to be clear that A | B C is A | (B C), not (A | B) C. 

Do not care comma so.